### PR TITLE
refactor: `TeamsChannels` to typescript

### PR DESCRIPTION
**Pull request by dougfabris**

### CODE REVIEW

File: apps/meteor/client/views/teams/contextualBar/channels/TeamsChannelsWithData.tsx [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+		e.preventDefault();
+
+		const handleClose = () => {
+			setModal(null);
+			reload();
+		};
+
+		setModal(() => <Component onClose={handleClose} teamId={teamId} />);
+	});
+};
```
Comment (tassoevan) [inline apps/meteor/client/views/teams/contextualBar/channels/TeamsChannelsWithData.tsx:31]: Should it be reusable? If so, move it to a new module.

File: apps/meteor/client/views/teams/contextualBar/channels/TeamsChannelsWithData.tsx [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+	const goToRoom = useCallback((room) => roomCoordinator.openRouteLink(room.t, room), []);
+	const handleBack = useCallback(() => setState({}), [setState]);
+	const viewRoom = useEffectEvent((room: IRoom) => {
+		goToRoom(room);
+	});
+
+	if (state?.tab === 'RoomInfo' && state?.rid) {
+		return <RoomInfo onClickBack={handleBack} onEnterRoom={goToRoom} resetState={() => setState({})} />;
+	}
```
Comment (tassoevan) [inline apps/meteor/client/views/teams/contextualBar/channels/TeamsChannelsWithData.tsx:71]: Why is this necessary?

**Review by tassoevan — CHANGES_REQUESTED** [review]

File: apps/meteor/client/views/teams/contextualBar/channels/TeamsChannelsWithData.tsx (hunk 2 above)
Comment (dougfabris) [reply apps/meteor/client/views/teams/contextualBar/channels/TeamsChannelsWithData.tsx:71]: to show the RoomInfo when clicking on the teams channels list

File: apps/meteor/client/views/teams/contextualBar/channels/TeamsChannelsWithData.tsx (hunk 1 above)
Comment (dougfabris) [reply apps/meteor/client/views/teams/contextualBar/channels/TeamsChannelsWithData.tsx:31]: I don't like this approach, so I dropped it

**Review by tassoevan — APPROVED** [review]
